src/utils/formatDataNumber.ts

```diff
@@ -31,7 +31,9 @@ export function formatDataNumber(
     return new Intl.NumberFormat('en-US', {
         minimumFractionDigits: minimumDecimals,
         maximumFractionDigits:
-            minimumDecimals >= formatDecimal ? Math.min(minimumDecimals, formatDecimal) + 1 : formatDecimal,
+            minimumDecimals >= formatDecimal && formatDecimal !== 0
+                ? Math.min(minimumDecimals, formatDecimal) + 1
+                : formatDecimal,
```

Comment:
Please verify throughout the app that this doesn't create unintended effects. After testing, please request Jacobs review